pkg/registry/core/rest/storage_core.go

```diff
@@ -342,8 +343,8 @@ type componentStatusStorage struct {
 func (s componentStatusStorage) serversToValidate() map[string]*componentstatus.Server {
 	// this is fragile, which assumes that the default port is being used
 	serversToValidate := map[string]*componentstatus.Server{
```

Comment:
I wouldn't switch this until these components remove the ability to serve insecurely. Only when there is zero chance of the previous configuration working on any deployment should we move this over to a new hard-coded configuration.